src/firebolt_db/firebolt_dialect.py

```diff
@@ -145,18 +145,32 @@ def create_connect_args(self, url: URL) -> Tuple[List, Dict]:
         """
         Build firebolt-sdk compatible connection arguments.
         URL format : firebolt://id:secret@host:port/db_name
+        For Core: firebolt://host:port/db_name?url=core_url
```

Comment:
host and port should not be a self-contained part of the connection string. Both of them as well as scheme are a part of the url parameter.